{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "680808c7",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "if (!requireNamespace(\"BiocManager\", quietly = TRUE)) install.packages(\"BiocManager\")\n",
    "BiocManager::install(c(\"TCGAbiolinks\", \"SummarizedExperiment\", \"dplyr\"))\n",
    "\n",
    "library(TCGAbiolinks)\n",
    "library(SummarizedExperiment)\n",
    "library(dplyr)\n",
    "\n",
    "query <- GDCquery(\n",
    "  project = \"TCGA-THCA\",\n",
    "  data.category = \"Transcriptome Profiling\",\n",
    "  data.type = \"Gene Expression Quantification\",\n",
    "  workflow.type = \"HTSeq - Counts\"\n",
    ")\n",
    "\n",
    "GDCdownload(query)\n",
    "se <- GDCprepare(query)   # SummarizedExperiment\n",
    "se"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "50ed7533",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "table(colData(se)$shortLetterCode)\n",
    "table(colData(se)$sample_type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "37fa2fa5",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "se_TN <- se[, colData(se)$sample_type %in% c(\"Primary Tumor\", \"Solid Tissue Normal\")]\n",
    "table(colData(se_TN)$sample_type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "20b044b4",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "clin <- GDCquery_clinic(project = \"TCGA-THCA\", type = \"clinical\")\n",
    "head(clin)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "30ed1ca6",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "sample_barcodes <- colnames(se_TN)\n",
    "patient_id <- substr(sample_barcodes, 1, 12)\n",
    "\n",
    "meta <- as.data.frame(colData(se_TN)) %>%\n",
    "  mutate(patient_id = patient_id)\n",
    "\n",
    "# Join clinique (selon la colonne patient/submitter id pr√©sente chez toi)\n",
    "# Souvent c'est \"submitter_id\" ou \"case_submitter_id\" selon la table.\n",
    "names(clin)\n",
    "\n",
    "# Exemple si la colonne s'appelle submitter_id :\n",
    "meta2 <- meta %>%\n",
    "  left_join(clin, by = c(\"patient_id\" = \"submitter_id\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "148d0649",
   "metadata": {
    "vscode": {
     "languageId": "r"
    }
   },
   "outputs": [],
   "source": [
    "meta_tumor <- meta2 %>%\n",
    "  filter(sample_type == \"Primary Tumor\") %>%\n",
    "  arrange(patient_id, sample_barcodes) %>%   # ordre stable\n",
    "  distinct(patient_id, .keep_all = TRUE)\n",
    "\n",
    "se_patient <- se_TN[, meta_tumor$barcode]   # selon le nom de la colonne barcode"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "name": "R"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
